app-emulation/wine-staging/wine-staging-6.0_rc4.ebuild

```diff
@@ -0,0 +1,663 @@
+		# Compile in subshell to prevent "Aborted" message
+		( $(tc-getCC) -O2 "${PATCHDIR}/files/builtin_ms_va_list.c" -o "${T}"/builtin_ms_va_list >/dev/null 2>&1)
+		if ! eend $?; then
+			eerror "This version of $(tc-getCC) does not support builtin_ms_va_list, can't enable 64-bit wine"
+			eerror
+			eerror "You need gcc-4.4+ or clang 3.8+ to build 64-bit wine"
+			eerror
+			return 1
+		fi
+	fi
+}
+
+wine_build_environment_check() {
+	[[ ${MERGE_TYPE} = "binary" ]] && return 0
+
+	if use abi_x86_64; then
+		if tc-is-gcc && [[ $(gcc-major-version) -lt 4 || ( $(gcc-major-version) -eq 4 && $(gcc-minor-version) -lt 4 ) ]]; then
+			eerror "You need gcc-4.4+ to compile 64-bit wine"
+			die
+		elif tc-is-clang && [[ $(clang-major-version) -lt 3 || ( $(clang-major-version) -eq 3 && $(clang-minor-version) -lt 8 ) ]]; then
+			eerror "You need clang-3.8+ to compile 64-bit wine"
+			die
+		fi
+	fi
+	if tc-is-gcc && [[ $(gcc-major-version) -eq 5 && $(gcc-minor-version) -le 3 ]]; then
+		ewarn "GCC-5.0-5.3 suffered from compiler bugs and are no longer supported by"
+		ewarn "Gentoo's Toolchain Team. If your ebuild fails the compiler checks in"
+		ewarn "the configure phase, either update your compiler or switch to <5.0 || >=5.4"
+	fi
+	if tc-is-gcc && [[ $(gcc-major-version) -eq 5 && $(gcc-minor-version) -eq 4 ]]; then
+		if has "-march=i686" ${CFLAGS} && ! has "-mtune=generic" ${CFLAGS}; then
+			ewarn "Compilation can hang with CFLAGS=\"-march=i686\".  You can temporarily work"
+			ewarn "around this by adding \"-mtune=generic\" to your CFLAGS for wine."
+			ewarn "See package.env in man 5 portage for more information on how to do this."
+			ewarn "See https://bugs.gentoo.org/show_bug.cgi?id=613128 for more details"
+		fi
+	fi
+
+	if use abi_x86_32 && use opencl && [[ "$(eselect opencl show 2> /dev/null)" == "intel" ]]; then
+		eerror "You cannot build wine with USE=opencl because intel-ocl-sdk is 64-bit only."
+		eerror "See https://bugs.gentoo.org/487864 for more details."
+		eerror
+		return 1
+	fi
+}
+
+wine_env_vcs_vars() {
+	local pn_live_var="${PN//[-+]/_}_LIVE_COMMIT"
+	local pn_live_val="${pn_live_var}"
+	eval pn_live_val='$'${pn_live_val}
+	if [[ ! -z ${pn_live_val} ]]; then
+		if use staging; then
+			eerror "Because of the multi-repo nature of ${MY_PN}, ${pn_live_var}"
+			eerror "cannot be used to set the commit. Instead, you may use the"
+			eerror "environment variables:"
+			eerror "  EGIT_OVERRIDE_COMMIT_WINE"
+			eerror "  EGIT_OVERRIDE_COMMIT_WINE_STAGING_WINE_STAGING"
+			eerror
+			return 1
+		fi
+	fi
+	if [[ ! -z ${EGIT_COMMIT} ]]; then
+		eerror "Commits must now be specified using the environment variables:"
+		eerror "  EGIT_OVERRIDE_COMMIT_WINE"
+		eerror "  EGIT_OVERRIDE_COMMIT_WINE_STAGING_WINE_STAGING"
+		eerror
+		return 1
+	fi
+}
+
+src_pretend() {
+	wine_build_environment_check || die
+
+	# Verify OSS support
+	if use oss && ! use kernel_FreeBSD; then
+		if ! has_version ">=media-sound/oss-4"; then
```

Comment:
Are there any other changes to the ebuild besides pkg_pretend->src_pretend?

Reply:
```
--- /usr/portage/app-emulation/wine-staging/wine-staging-5.22-r2.ebuild 2020-12-05 09:24:53.880476216 +0000
+++ wine-staging-6.0_rc4.ebuild 2020-12-31 22:21:52.447298487 +0000
@@ -7,11 +7,11 @@
 PLOCALE_BACKUP="en"
 
 inherit autotools eapi7-ver estack eutils flag-o-matic gnome2-utils l10n multilib multilib-minimal pax-utils toolchain-funcs virtualx xdg-utils
-
 MY_PN="${PN%%-*}"
-MY_P="${MY_PN}-${PV}"
+MY_PV="${PV/_/-}"
+MY_P="${MY_PN}-${MY_PV}"
 
-if [[ ${PV} == "9999" ]] ; then
+if [[ ${MY_PV} == "9999" ]] ; then
        EGIT_REPO_URI="https://source.winehq.org/git/wine.git"
        EGIT_BRANCH="master"
        inherit git-r3
@@ -19,12 +19,16 @@
        #KEYWORDS=""
 else
        MAJOR_V=$(ver_cut 1)
-       SRC_URI="https://dl.winehq.org/wine/source/${MAJOR_V}.x/${MY_P}.tar.xz"
+       MINOR_V=$(ver_cut 2)
+    if [[ ${MINOR_V} != "0" ]] ; then
+        MINOR_V="x"
+    fi
+       SRC_URI="https://dl.winehq.org/wine/source/${MAJOR_V}.${MINOR_V}/${MY_P}.tar.xz"
        KEYWORDS="-* ~amd64 ~x86"
 fi
 S="${WORKDIR}/${MY_P}"
 
-STAGING_P="wine-staging-${PV}"
+STAGING_P="wine-staging-${MY_PV}"
 STAGING_DIR="${WORKDIR}/${STAGING_P}"
 GWP_V="20200523"
 PATCHDIR="${WORKDIR}/gentoo-wine-patches"
@@ -35,15 +39,15 @@
        https://dev.gentoo.org/~sarnex/distfiles/wine/gentoo-wine-patches-${GWP_V}.tar.xz
 "
 
-if [[ ${PV} == "9999" ]] ; then
+if [[ ${MY_PV} == "9999" ]] ; then
        STAGING_EGIT_REPO_URI="https://github.com/wine-staging/wine-staging.git"
 else
        SRC_URI="${SRC_URI}
-       staging? ( https://github.com/wine-staging/wine-staging/archive/v${PV}.tar.gz -> ${STAGING_P}.tar.gz )"
+       staging? ( https://github.com/wine-staging/wine-staging/archive/v${MY_PV}.tar.gz -> ${STAGING_P}.tar.gz )"
 fi
 
 LICENSE="LGPL-2.1"
-SLOT="${PV}"
+SLOT="${MY_PV}"
 IUSE="+abi_x86_32 +abi_x86_64 +alsa capi cups custom-cflags dos elibc_glibc +faudio +fontconfig +gcrypt +gecko gphoto2 gsm gssapi gstreamer +jpeg kerberos kernel_FreeBSD +lcms ldap mingw +mono mp3 netapi nls odbc openal opencl +opengl osmesa oss +perl pcap pipelight +png prelink pulseaudio +realtime +run-exes samba scanner sdl selinux +ssl staging test themes +threads +truetype udev +udisks +unwind v4l vaapi vkd3d vulkan +X +xcomposite xinerama +xml"
 REQUIRED_USE="|| ( abi_x86_32 abi_x86_64 )
        X? ( truetype )
@@ -171,7 +175,7 @@
 PATCHES_BIN=()
 
 # https://bugs.gentoo.org/show_bug.cgi?id=635222
-if [[ ${#PATCHES_BIN[@]} -ge 1 ]] || [[ ${PV} == 9999 ]]; then
+if [[ ${#PATCHES_BIN[@]} -ge 1 ]] || [[ ${MY_PV} == 9999 ]]; then
        DEPEND+=" dev-util/patchbin"
 fi
 
@@ -284,7 +288,7 @@
        fi
 }
 
-pkg_pretend() {
+src_pretend() {
        wine_build_environment_check || die
 
        # Verify OSS support
@@ -330,7 +334,7 @@
        wine_build_environment_check || die
        wine_env_vcs_vars || die
 
-       WINE_VARIANT="${PN#wine}-${PV}"
+       WINE_VARIANT="${PN#wine}-${MY_PV}"
        WINE_VARIANT="${WINE_VARIANT#-}"
 
        MY_PREFIX="${EPREFIX}/usr/lib/wine-${WINE_VARIANT}"
@@ -344,7 +348,7 @@
 }
 
 src_unpack() {
-       if [[ ${PV} == "9999" ]] ; then
+       if [[ ${MY_PV} == "9999" ]] ; then
                EGIT_CHECKOUT_DIR="${S}" git-r3_src_unpack
                if use staging; then
                        local CURRENT_WINE_COMMIT=${EGIT_VERSION}
@@ -614,12 +618,12 @@
 }
 
 pkg_postinst() {
-       eselect wine register ${P}
+       eselect wine register ${PN}-${MY_PV}
        if [[ ${PN} == "wine-vanilla" ]]; then
-               eselect wine register --vanilla ${P} || die
+               eselect wine register --vanilla ${PN}-${MY_PV} || die
        else
                if use staging; then
-                       eselect wine register --staging ${P} || die
+                       eselect wine register --staging ${PN}-${MY_PV} || die
                fi
        fi
 
@@ -642,12 +646,12 @@
 }
 
 pkg_prerm() {
-       eselect wine deregister ${P}
+       eselect wine deregister ${PN}-${MY_PV}
        if [[ ${PN} == "wine-vanilla" ]]; then
-               eselect wine deregister --vanilla ${P} || die
+               eselect wine deregister --vanilla ${PN}-${MY_PV} || die
        else
                if use staging; then
-                       eselect wine deregister --staging ${P} || die
+                       eselect wine deregister --staging  ${PN}-${MY_PV}|| die
                fi
        fi
```

Reply:
That's the diff between 5.22-r2 and my ebuild